### PR TITLE
feat: add time range events filter

### PR DESCRIPTION
**Pull request by eliobischof**
### API Changes

- Deprecates the creationDate filter in favor of the oneof From filter
- The oneof From filter behaves exactly the same as the creationDate filter
- The oneof Range filter enables specifiying a since and until date

### Console Changes

- Seconds are displayed in the events creation dates
- Enables filtering in console by time range with second precision
- Creation Date is sorted instead of sequence

image

### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: console/src/app/modules/filter-events/filter-events.component.html [hunk 1]
```diff
@@ -145,55 +145,81 @@
...
+          </mat-radio-group>
+          <div class="filter-events-sub">
+            <ng-container *ngIf="creationDateFilterType!.value == CreationDateFilterType.FROM">
+              <mat-form-field class="datetime-input">
+                <input
+                  class="datetime"
+                  matInput
+                  type="datetime-local"
+                  name="creationDateFrom"
+                  [value]="creationDateFrom!.value | date: 'yyyy-MM-ddTHH:mm:ss'"
```
Comment (peintnermax) [inline console/src/app/modules/filter-events/filter-events.component.html:178]:
```suggestion
                  [value]="creationDateFrom!.value | date: 'yyyy-MM-dd HH:mm:ss'"
```

File: console/src/app/modules/filter-events/filter-events.component.html [hunk 2]
```diff
@@ -145,55 +145,81 @@
...
+            <ng-container *ngIf="creationDateFilterType!.value == CreationDateFilterType.RANGE">
+              <div class="datetime-range">
+                <mat-form-field class="datetime-input">
+                  <cnsl-label
+                    >{{ 'IAM.EVENTS.FILTERS.CREATIONDATE.LABEL_SINCE' | translate }}
+                    <input
+                      matInput
+                      type="datetime-local"
+                      name="creationDateSince"
+                      [value]="creationDateSince!.value | date: 'yyyy-MM-ddTHH:mm:ss'"
```
Comment (peintnermax) [inline console/src/app/modules/filter-events/filter-events.component.html:192]:
```suggestion
                      [value]="creationDateSince!.value | date: 'yyyy-MM-dd HH:mm:ss'"
```

File: console/src/app/modules/filter-events/filter-events.component.html [hunk 3]
```diff
@@ -145,55 +145,81 @@
...
+                  </cnsl-label>
+                </mat-form-field>
+                <mat-form-field class="datetime-input">
+                  <cnsl-label
+                    >{{ 'IAM.EVENTS.FILTERS.CREATIONDATE.LABEL_UNTIL' | translate }}
+                    <input
+                      matInput
+                      type="datetime-local"
+                      name="creationDateUntil"
+                      [value]="creationDateUntil!.value | date: 'yyyy-MM-ddTHH:mm:ss'"
```
Comment (peintnermax) [inline console/src/app/modules/filter-events/filter-events.component.html:204]:
```suggestion
                      [value]="creationDateUntil!.value | date: 'yyyy-MM-dd HH:mm:ss'"
```

**Comment by eliobischof** [comment]
@adlerhurst could you please review the api changes (.go and .proto)
@peintnermax could you please review the rest.
thx

**Review by peintnermax — CHANGES_REQUESTED** [review]

**Review by peintnermax — DISMISSED** [review]

File: internal/query/event.go [hunk 4]
```diff
@@ -66,7 +66,7 @@ func filterAuditLogRetention(ctx context.Context, auditLogRetention time.Duratio
 	if callTime.IsZero() {
 		callTime = time.Now()
 	}
-	oldestAllowed := callTime.Add(-auditLogRetention)
+	oldestAllowed := callTime.UTC().Add(-auditLogRetention)
```
Comment (adlerhurst) [inline internal/query/event.go:69]: why did you add this?

Comment (eliobischof) [reply internal/query/event.go]: I carefully tested it again, and you are right. I removed it.
